xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/validation/livevalidation_prototype.js

```diff
@@ -171,18 +169,36 @@ LiveValidation.prototype = {
       }
     }
     this.validations = [];
-	this.removeMessageAndFieldClass();
+	this.removeFieldClass();
   },
   
   /**
-   *	adds a validation to perform to a LiveValidation object
+   *	adds a validation to perform on a LiveValidation object
    *
    *	@param validationFunction {Function} - validation function to be used (ie Validate.Presence )
    *	@param validationParamsObj {Object} - parameters for doing the validation, if wanted or necessary
    *    @return {Object} - the LiveValidation object itself so that calls can be chained
    */
   add: function(validationFunction, validationParamsObj){
-    this.validations.push( { type: validationFunction, params: validationParamsObj || {} } );
+    let validationMessageHolder;
+    /* The identifier helps us give specific behaviour to different kinds of validations. The user expectations change
+     * depending on the validation type, and this identifier makes sure we can retrieve the right message from the set
+     * of messages in the template. */
+    if (validationParamsObj.identifier) {
+      validationMessageHolder = this.options.insertAfterWhatNode.up().querySelector("." + validationParamsObj.identifier);
+      /* We use the failure message as a fallback for the success message for regexes. This ensures that the regex stays
+       * shown to the user at all times. */
+      if (validationParamsObj.identifier.includes("regex") && validationParamsObj.validMessage == null) {
+        validationParamsObj.validMessage = validationParamsObj.failureMessage;
+      }
+    } else {
+      /* If we don't have an identifier for the validation, this is a legacy use of the livevalidation.
+        We create a message holder from scratch to make it compatible with the way the new implementation works. */
+      validationMessageHolder = this.createMessageSpan();
+      this.options.insertAfterWhatNode.up().appendChild(validationMessageHolder);
+    }
+    /* We add one validation object to the list of validations for the current field. */
+    this.validations.push( { type: validationFunction, params: validationParamsObj || {}, messageHolder: validationMessageHolder } );
```

Comment:
Hmmm we're suppose to try to migrate out of prototype... a bit a pity to add new code using it.

Reply:
I tried to keep it as simple as possible while keeping backward compatibility for all the users that extended the form with their own content. 

Do we already have a ticket open to rework the whole livevalidation system to work in native js + jquery?

IMO the scope of this PR is already large enough, I'd rather merge a rework of the livevalidation js in another commit.

Reply:
> Do we already have a ticket open to rework the whole livevalidation system to work in native js + jquery?

I don't think so.

> IMO the scope of this PR is already large enough, I'd rather merge a rework of the livevalidation js in another commit.

Ok fine with me.

Reply:
Created https://jira.xwiki.org/browse/XWIKI-22596 :+1: